src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -283,15 +283,20 @@ address MacroAssembler::target_addr_for_insn(address insn_addr, unsigned insn) {
     return address(uint64_t(Instruction_aarch64::extract(insns[0], 20, 5))
                    + (uint64_t(Instruction_aarch64::extract(insns[1], 20, 5)) << 16)
                    + (uint64_t(Instruction_aarch64::extract(insns[2], 20, 5)) << 32));
-  } else if (Instruction_aarch64::extract(insn, 31, 22) == 0b1011100101 &&
```

Comment:
It must be a check for safePoint poll: ldrw zr, polling_page. Technically, the target address for this instruction is zero and the function is correct. I am not sure Parfait is right in his complaints. After all, does this code ever works? That is, if you remove it, will the SholdNotReachHere below fire?

Reply:
I don't think the safepoint poll is coming thru this code path. I ran this fix
through Mach5 Tier[1-6] and did not see any failures caused by hitting
a `ShouldNotReachHere()` call. It's hard to imagine that Tier[1-6] testing
wouldn't exercise the safepoint polling code paths.

I've also verified that the Parfait tool no longer issues the complaints.

I will poke around and see if I can determine what the safepoint polling
is doing.

Reply:
@bulasevich - I've made another pass through the code and checked the
calls to `MacroAssembler::target_addr_for_insn()` and I don't see any of
those calls in areas related to safepoint polling. From the other POV, I
took a look at the aarch64 safepoint polling code and I don't see a way
into `MacroAssembler::target_addr_for_insn()` .

It is entirely possible that I'm missing something here and would appreciate
it if you could sanity check my conclusions here.

Reply:
I believe this code is dead, in both `MacroAssembler::pd_patch_instruction()` and `MacroAssembler::target_addr_for_insn()`, since the poll read is PIC, we are not patching and should not need to acquire any target to update.

The associated `fix_relocation_after_move()` will not invoke any of the above since the `maybe_cpool_ref()` condition is used:
```
void poll_Relocation::fix_relocation_after_move(const CodeBuffer* src, CodeBuffer* dest) {
  if (NativeInstruction::maybe_cpool_ref(addr())) {
    address old_addr = old_addr_for(addr(), src, dest);
    MacroAssembler::pd_patch_instruction(addr(), MacroAssembler::target_addr_for_insn(old_addr));
  }
}
```
Should be sufficient to simply remove these cases, possibly you could move an assert into the "unreachable" branch (for "precision").
```
} else {
  assert(!NativeInstruction::is_ldrw_to_zr(<addr>), "Unexpected poll read");
  ShouldNotReachHere();
}
```
(The code above in `fix_relocation_after_move()` might also be dead...)

Chip in @theRealAph ?

Reply:
@phedlin - Thanks for chiming in on this review thread.

Reply:
I wondered if target_addr_for_insn ever gets the ldrw_to_zr instruction. 
Anyway, I am Ok to move it to target_addr_for_insn_allow_null_ret (target_addr_for_insn_or_null?).
Thank you.

Reply:
Since we moved to using per-thread safepoints, I don't think we'll ever see a reloc against a safepoint poll. I've just looked through the AArch64 back end, and as far as I can see the old safepoint code is gone.

Reply:
> @bulasevich - I've made another pass through the code and checked the calls to `MacroAssembler::target_addr_for_insn()` and I don't see any of those calls in areas related to safepoint polling. From the other POV, I took a look at the aarch64 safepoint polling code and I don't see a way into `MacroAssembler::target_addr_for_insn()` .
> 
> It is entirely possible that I'm missing something here and would appreciate it if you could sanity check my conclusions here.

I think you're right. As far as I can see, this code is dead since 
8253180: ZGC: Implementation of JEP 376: ZGC: Concurrent Thread-Stack Processing

which took out the code that generated safepoint relocs, but not the code here that relocated them.

Reply:
I've filed the following new RFE to track the dead compiler code:

    JDK-8280283 dead compiler code found during the JDK-8272058 code review
    https://bugs.openjdk.java.net/browse/JDK-8280283